mantidproject/mantid
# Add gui extension for extended trans w beam stop in sans

Fixes #12973
### For testing

In this ticket we want to add a GUI interface for the chagnes made in #10192 .

The test data can be found here: \olympic\Babylon5\Scratch\Anton\Testing\12973_Add_GUI_For_Extend_Trans_Command\Sandbox
The directory above contains a windows installer: \olympic\Babylon5\Scratch\Anton\Testing\12973_Add_GUI_For_Extend_Trans_Command

We want to test several user file scenarios:
1.  user_file_M3_old.txt
2.  user_file_with_bs_RADIUS.txt
3.  user_file_with_bs_ROI.txt
4.  user_file_with_bs_ROI_and_RADIUS.txt
5.  user_file_with_bs_ROI_and_RADIUS_and_Monitor.txt

The instrument is SANS2DTubes.

The data sets which need to be loaded are
- Sample Scattering: 23785-add.nxs
- Sample Transmission: 23763-add.nxs
- Sample Direct: 23764-add.nxs
- Can Scattering: 23784-add.nxs
- Can Transmission:  23764-add.nxs
- Can Direct:  23764-add.nxs

**Test button logic**
1. Open the ISIS SANS GUI, make sure that SANS2DTubes is set and load the user file: user_file_M3_old.txt
2. Go to the Reduction Settings tab. We are only intereseted in the section Transmission Settings
   -  Confirm that only M3 is checked.
   -  Confirm that only the line edit field "shift" is enabled.
3. Check the M4 checkbox
   - Confirm that M3 get disabled
4. Check the Radius checkbox
   - Confirm that M4 is unchecked
   - Confirm that the "Shift" line edit field is disabled
   - Confirm that the Radius line edit field and the Masking files line edit field are enabled
5. Check the ROI files checkbox
   - Confirm that the Radisu checkbox stays checked
   - Confirm that the ROI files checkbox is checked and the ROI files line editr field is enabled
6. Type something into the Radius field and uncheck the Radius checkbox.
   - Confirm that the masking files line edit field is still enabled
   - Confirm that the Radius line edit field is disabled
7. Uncheck the ROI files checkbox
   - Confirm that the ROI files line edit field is disabled
   - Confirm that the Masking files line editr field is disabled

**Test loading of different files sets the checkboxes and fields correctly**
1. Open the ISIS SANS GUI, make sure that SANS2DTubes is set and load the user file: user_file_M3_old.txt
   - Confirm that only the M3 checkbox is set
2. Load the user file: user_file_with_bs_RADIUS.txt
   - Confirm that the Radius checkbox is checked
   - Confirm that the Radius line edit field is set to 30
3. Load the user file: user_file_with_bs_ROI.txt
   - Confirm that the ROI files checkbox is checked
   - Confirm that the ROI files line edit field contains TRANS_ROI.xml
   - Confirm that the Masking files line edit field contains TRANS_MASK.xml
4. Load the user file: user_file_with_bs_ROI_and_RADIUS.txt
   - Confirm that the ROI files and Radius checkboxes is checked
   - Confirm that the ROI files line edit field contains TRANS_ROI.xml
   - Confirm that the Radius line edit field is set to 30
   - Confirm that the Masking files line edit field contains TRANS_MASK.xml
5. Load the user file: user_file_with_bs_ROI_and_RADIUS_and_Monitor.txt
   - Confirm that M3 is checked
   - Confirm that the ROI files line edit field contains TRANS_ROI.xml but is disabled.
   - Confirm that the Radius line edit field is set to 30 but is disabled.
   - Confirm that the Masking files line edit field contains TRANS_MASK.xml but is disabled.
     Confirm that the "Shift" line edit field is enabled

In order to check if the GUI settings are passed on we need to compare workspaces.

**Test that setting ROI file is picked up**
1. Open the ISIS SANS GUI, make sure that SANS2DTubes is set and load the user file: user_file_with_bs_RADIUS.txt
2. Load the sample files as described above
3. Check the ROI file setting, set TRANS_ROI.xml in the input; set TRANS_MASK.xml in the Masking file input
4. Apply 1D reduce
5. Rename the file 23775_trans_sample_1.75_16_5_unfitted to roi_reference
6. Load the user file:  user_file_with_bs_ROI_and_RADIUS.txt
7. Load the sample files again
8. Apply 1D reduce
   9 Compare 23775_trans_sample_1.75_16_5_unfitted with radius_reference by plotting them on top of each other
   - Confirm that they match
